extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityUtils.java

```diff
@@ -18,4 +23,26 @@ public static AuthenticationRequest setRoutingContextAttribute(AuthenticationReq
     public static RoutingContext getRoutingContextAttribute(AuthenticationRequest request) {
         return request.getAttribute(ROUTING_CONTEXT_ATTRIBUTE);
     }
+
+    /**
+     * Add {@link RoutingContext} to Vert.x duplicated context local data.
+     */
+    public static void setRoutingContextAttribute(RoutingContext event) {
+        final Context context = Vertx.currentContext();
+        if (isSafeToUseContext(context)) {
+            context.putLocal(RoutingContext.class.getName(), event);
```

Comment:
This will blindly set it always, even if present already: this is what we need? or we need avoiding setting it again if already present, and failing if what's present is different (or any other behaviour we want to enforce).

Sadly `ctx::putLocal` is  a `ConcurrentHashMap::put` which can become way cheaper if it doesn't always happen (unless is the semantic we requires, or we know that is always transitioning the value from being absent to the new value).

Reply:
I think relation of RoutingContext to duplicated context is 1:N (but at this point it will most likely be 1:1, just in case someone create a new one later?) so it will always be the same. And it is not there otherwise we would not bother to add it there. I can't presume if situation can change in the future.

Sorry I don't have a good answer for this, but I appreciate your input. I'd like to always add it there because right now we are talking augmentors, but users can't access it anywhere during authentication and authorization (that is simplified - list of cases is also complicated)

Reply:
Quick check @franz1981  - do you think it makes sense to always run `get != null` before adding it even though it won't be the case unless something changes in the future? I can add it.

Reply:
To me it's fine as it is as long as the semantic for who use it is clear and doc somewhere :+1:

Reply:
Well I thought that would be answer on not executing unnecessary `ConcurrentHashMap::put`, but as said, unless someone starts to add it, this code runs pretty early so it won't be there.

Reply:
Okay, so I have added extra condition to address your comment.